it/common/src/main/java/org/apache/beam/it/common/utils/ResourceManagerUtils.java

```diff
@@ -0,0 +1,208 @@
+        illegalChars.matcher(baseString.toLowerCase()).replaceAll(replaceChar);
+
+    // finally, append the date/time and return the substring that does not exceed the length limit
+    LocalDateTime localDateTime = LocalDateTime.now(ZoneId.of(TIME_ZONE));
+    String timeAddOn = localDateTime.format(timeFormat);
+    return illegalCharsRemoved.subSequence(
+            0, min(targetLength - timeAddOn.length() - 1, illegalCharsRemoved.length()))
+        + replaceChar
+        + localDateTime.format(timeFormat);
+  }
+
+  /** Generates random letter for padding. */
+  public static char generatePadding() {
+    Random random = new Random();
+    return (char) ('a' + random.nextInt(26));
+  }
+
+  /**
+   * Checks whether the given project ID is valid according to GCP constraints.
+   *
+   * @param idToCheck the project ID to check.
+   * @throws IllegalArgumentException if the project ID is invalid.
+   */
+  public static void checkValidProjectId(String idToCheck) {
+    if (idToCheck.length() < MIN_PROJECT_ID_LENGTH) {
+      throw new IllegalArgumentException("Project ID " + idToCheck + " cannot be empty.");
+    }
+    if (idToCheck.length() > MAX_PROJECT_ID_LENGTH) {
+      throw new IllegalArgumentException(
+          "Project ID "
+              + idToCheck
+              + " cannot be longer than "
+              + MAX_PROJECT_ID_LENGTH
+              + " characters.");
+    }
+    if (ILLEGAL_PROJECT_CHARS.matcher(idToCheck).find()) {
+      throw new IllegalArgumentException(
+          "Project ID "
+              + idToCheck
+              + " is not a valid ID. Only letters, numbers, hyphens, single quotes, colon, dot and"
+              + " exclamation points are allowed.");
+    }
+  }
+
+  /**
+   * Cleanup Resources from the given ResourceManagers. It will guarantee that all the cleanups are
+   * invoked, but still throws / bubbles the first exception at the end if something went wrong.
+   *
+   * @param managers Varargs of the managers to clean
+   */
+  public static void cleanResources(ResourceManager... managers) {
+
+    if (managers == null || managers.length == 0) {
+      return;
+    }
+
+    Exception bubbleException = null;
+
+    for (ResourceManager manager : managers) {
+      if (manager == null) {
+        continue;
+      }
+      try {
+        LOG.info("Cleaning up resource manager {}", manager.getClass().getSimpleName());
+        manager.cleanupAll();
+      } catch (Exception e) {
+        LOG.error("Error cleaning the resource manager {}", manager.getClass().getSimpleName());
+        if (bubbleException == null) {
+          bubbleException = e;
+        }
+      }
+    }
+
+    if (bubbleException != null) {
+      throw new RuntimeException("Error cleaning up resources", bubbleException);
+    }
+  }
+
+  /**
+   * Generates a password using random characters for tests.
+   *
+   * <p>Note: The password generated is not cryptographically secure and should only be used in
```

Comment:
I missed this in the epic review. Because random people on the internet are doing generic searches for vulnerabilities, this will get reported over and over even if it is not a real vulnerability. Can you just change it to be something that _is_ secure, or find a way to eliminate the need for it?